modules/portal/portal-api/src/main/java/com/enonic/xp/portal/handler/EndpointHandler.java

```diff
@@ -10,40 +13,52 @@
 public abstract class EndpointHandler
     extends BaseWebHandler
 {
-    private final String path;
+    public static final String ENDPOINT_TYPE_PREFIX = "/_/";
 
-    private final String pathPrefix;
+    private final Set<String> endpoints;
 
     public EndpointHandler( final String endpointType )
     {
         this( HttpMethod.standard(), endpointType );
     }
 
-    public EndpointHandler( final EnumSet<HttpMethod> methodsAllowed, final String endpointType )
+    public EndpointHandler( final EnumSet<HttpMethod> methodsAllowed, final String endpoint )
     {
         super( methodsAllowed );
-        this.path = "/_/" + endpointType;
-        this.pathPrefix = path + "/";
+        this.endpoints = Set.of( endpoint );
+    }
+
+    public EndpointHandler( final int order, final EnumSet<HttpMethod> methodsAllowed, final String... endpoints )
+    {
+        super( order, methodsAllowed );
+        this.endpoints = Arrays.stream( endpoints ).collect( Collectors.toSet() );
     }
 
     @Override
     public boolean canHandle( final WebRequest req )
     {
         final String endpointPath = req.getEndpointPath();
-        return endpointPath != null && ( endpointPath.startsWith( pathPrefix ) || endpointPath.equals( path ) );
+        return endpointPath != null && endpoints.stream()
+            .anyMatch(
+                s -> endpointPath.startsWith( ENDPOINT_TYPE_PREFIX + s + "/" ) || ( ENDPOINT_TYPE_PREFIX + s ).equals( endpointPath ) );
```

Comment:
Before code vas garbage-free, now it creates a few string objects for every check.

Reply:
Maybe it is better instead to create multiple `ServiceMappingHandler`s